### PR TITLE
chore: fix ruby version and improve CI build time

### PR DESCRIPTION
**Pull request by sethkfman**
## **Description**

This PR removes build steps not needed for building iOS applications and improves the build time for iOS by approx 2 min.

Build step removed for prod, flask, qa iOS builds:
- Ruby and cocoa pod utility install
- Cocoapod install

Ruby version for 3.1.4 is also upgrade which is the default for Bitrise.  

## **Related issues**

Fixes:

## **Manual testing steps**

1. Run E2E successfully

### **Before**

Screenshot 2023-12-20 at 4 52 53 PM

### **After**

Screenshot 2023-12-20 at 4 55 31 PM

Passing Builds
 iOS Flask - https://app.bitrise.io/build/6478972a-e679-4ef2-893c-e07416051f64
 iOS E2E Build - https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/e532ef49-e2b2-48ec-9bff-60e3c975a16b

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by tommasini — DISMISSED** [review]
LGTM!

File: bitrise.yml [hunk 1]
```diff
@@ -1050,6 +1051,12 @@ workflows:
             - build_short_version_string: $FLASK_VERSION_NAME
             - build_version: $FLASK_VERSION_NUMBER
             - plist_path: $PROJECT_LOCATION_IOS/MetaMask/MetaMask-Flask-Info.plist
+      - script:          
+          inputs:
+            - content: |-
+                asdf install ruby $RUBY_VERSION
```
Comment (Gudahtt) [inline bitrise.yml:1057]: Where is this environment variable set?

File: .ruby-version [hunk 2]
```diff
@@ -1 +1 @@
-3.0.0
+3.1.4
```
Comment (Gudahtt) [inline .ruby-version:1]: The PR description mentions 3.1.6 but this says 3.1.4. Could you clarify which one is the Bitrise default, and whether this discrepency exists or is intended?

Comment (sethkfman) [reply .ruby-version:1]: Typo I was moving too fast

File: bitrise.yml (hunk 1 above)
Comment (sethkfman) [reply bitrise.yml]: I missed removing this step. Here is a new build to confirm fix: https://app.bitrise.io/build/6478972a-e679-4ef2-893c-e07416051f64

File: bitrise.yml [hunk 3]
```diff
@@ -180,6 +180,15 @@ workflows:
           inputs:
             - command: setup
           title: Yarn Setup
+      - yarn@0:
```
Comment (Gudahtt) [inline bitrise.yml:183]: Were these steps intentionally restored? I don't see this referenced in the PR description.

It looks like they were recently removed in https://github.com/MetaMask/metamask-mobile/pull/7975 (it's unclear to me why)

**Review by Gudahtt — APPROVED** [review]

**Pull request merged**